LoRaEngine/LoraKeysManagerFacade/EdgeDeviceGetter.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using LoRaTools;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Logging;
+
+    public class EdgeDeviceGetter : IEdgeDeviceGetter
+    {
+        private readonly IDeviceRegistryManager registryManager;
+        private readonly ILoRaDeviceCacheStore cacheStore;
+        private readonly ILogger<EdgeDeviceGetter> logger;
+        private DateTimeOffset? lastUpdateTime;
+
+        public EdgeDeviceGetter(IDeviceRegistryManager registryManager,
+                                ILoRaDeviceCacheStore cacheStore,
+                                ILogger<EdgeDeviceGetter> logger)
+        {
+            this.registryManager = registryManager;
+            this.cacheStore = cacheStore;
+            this.logger = logger;
+        }
+
+#pragma warning disable IDE0060 // Remove unused parameter. Kept here for future improvements of RegistryManager
```

Comment:
Do we want to have a work item to keep this under radar?

Reply:
We should not need a work item here. Ideally if a future version of the RegistryManager will support cancellation tokens, as soon as we upgrade the dependency, the build should fail as we are not propagating the cancellation token (as per https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2016)